### ISSUE TITLE
get_correlations_ts float32 JSON not serialisable error

### ISSUE THREAD

**Issue by cdeil** [man-group/dtale#434]
I'm getting this error.
Does it make sense to you from the traceback?
Or should I try to create a minimal reproducer?
This is with a dataframe mostly consisting of float32 data.

```
Traceback (most recent call last):
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/dtale/views.py", line 83, in _handle_exceptions
    return func(*args, **kwargs)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/dtale/views.py", line 2845, in get_correlations_ts
    return jsonify(return_data)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/dtale/utils.py", line 599, in jsonify
    return _jsonify(return_data)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/flask/json/__init__.py", line 370, in jsonify
    dumps(data, indent=indent, separators=separators) + "\n",
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/flask/json/__init__.py", line 211, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/simplejson/__init__.py", line 398, in dumps
    return cls(
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/simplejson/encoder.py", line 296, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/simplejson/encoder.py", line 378, in iterencode
    return _iterencode(o, 0)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/flask/json/__init__.py", line 100, in default
    return _json.JSONEncoder.default(self, o)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/simplejson/encoder.py", line 272, in default
    raise TypeError('Object of type %s is not JSON serializable' %
TypeError: Object of type float32 is not JSON serializable
```
<img width="1410" alt="Screenshot 2021-02-16 at 22 27 05" src="https://user-images.githubusercontent.com/852409/108123419-1b4f5c80-70a6-11eb-841f-e9a072f4843b.png">

**Comment by aschonfeld** [comment]
So this would happen if a float32 value doesn’t get converted to a float when sending it back the browser. Could be caused by a NaN, since those are technically floats. I’ll do my best to fiddle around with some data on my end.

**Comment by aschonfeld** [comment]
Just from a review of the code it looks like I'm dropping NaN values from the data that goes into the chart but I wonder if the PPS (predictive power score) data is sneaking some NaN values in there.

**Comment by aschonfeld** [comment]
I going to add some extra format handling around the PPS data and hopefully that does the trick.  I'll include it in my next release and I'll give you a heads up when its out

**Comment by cdeil** [comment]
Thanks! Just to confirm, yes, I do start with float32 data with various amounts of NaNs in each column.

**Comment by aschonfeld** [comment]
@cdeil I've added some extra handling for the PPS data in the latest release v1.36.0, hopefully it solves your issue.  Please let me know if it still fails.

**Comment by cdeil** [comment]
@aschonfeld  - The issue is resolved, thank you very much!

I still do see this error in the console, but didn't notice incorrect behaviour of the app. Not sure if it's something that should be fixed or that I should ignore, in any case it's completely unrelated to the issue above.
```
2021-02-22 13:48:12,075 - ERROR    - Exception occurred while processing request: min() arg is an empty sequence
Traceback (most recent call last):
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/dtale/views.py", line 85, in _handle_exceptions
    return func(*args, **kwargs)
  File "/Users/cdeil/opt/anaconda3/envs/hpred_free_lime/lib/python3.8/site-packages/dtale/views.py", line 2907, in get_scatter
    idx = min(data[data[date_col] == date].index) + 1
ValueError: min() arg is an empty sequence
```
Let me know if opening an issue about this or trying to create a minimal reproducer would be useful to you.

**Comment by aschonfeld** [comment]
I’ll reopen this. If theres any way you could create either a snippet of code or just a small data dump (scrubbed of anything personal of course) that reproduces it that would be awesome. I know how to fix this but i’m just wondering if theres other stuff going on with your sample that I just havent thought of yet.  Thanks so much for all your help

**Comment by aschonfeld** [comment]
So I just pushed a fix that I'm hoping will solve your other issue.  If you clone master and run a `python setup.py develop` on your local environment you can test it early.  Otherwise I'll let you know when it gets released.

**Comment by cdeil** [comment]
I ran into #443 trying to install and use the dev version.

Before attempting to update, I didn't manage to create a minimal reproducer, not sure which columns and values in my large dataframe were responsible. I can try again with the same dataset after the next release.

This does give an exception about the all-NaN column, not sure if that should give an exception or is something that should be handled in the code and UI in a non-exception way.
```
datetime,aaa,bbb
2019-01-01 06:01:00+01:00,100,NULL
2019-01-01 06:02:00+01:00,101,NULL
2019-01-01 06:04:00+01:00,100,NULL
```

**Comment by cdeil** [comment]
I was reading the example like this, creating an index with name "datetime".
```python
import pandas as pd
import dtale
df = pd.read_csv("dtale_reproducer.csv").set_index("datetime")
dtale.show(df, open_browser=True)
```

**Comment by aschonfeld** [comment]
oh, it looks like you're dealing with timestamps and not dates.  I think if you drop the `datetime` column it might work fine.

Well, "work fine" in the sense that it will correlate all the values in column `aaa` against all the values in column `bbb`.  If you needed the date for something then I need to do some digging on whether my code will handle timestamps.

**Comment by cdeil** [comment]
Yes, I have sensor data with timestamps in my dataframe index with `df.index.name == "datetime"`.
The type is https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DatetimeIndex.html

**Comment by aschonfeld** [comment]
It looks like my current code does work with timestamps (although I need to fix a bug with the start date on the scatter)

https://user-images.githubusercontent.com/11547371/108913375-db353000-75f7-11eb-9c29-8b3f3ecd2291.mov

**Comment by cdeil** [comment]
For me, everything works OK now with the latest version on master. Certainly the original error about float32 JSON encoding is gone. Closing this issue.

Thank you for trying out timestamp as index, and fixing other issues if you notice them. There were some things like the "ValueError: min() arg is an empty sequence" that for me only showed up with some specific row and column selections, possibly related to some or all `NaN` columns, or non-unique or non-monotonic `DateTimeIndex`, I'm not sure, I didn't to boil it down to minimal reproducer examples.

Anyways, all seems good now. :-)

**Comment by aschonfeld** [comment]
Yea I think the `ValueError: min()` got fixed with my latest change to master.  So at the time it was definitely valid.  I'll add the fix for the start date display in the scatter and let you know when a new version has been released.  Might  be a little while though since I'm working thru internationalization at the moment.

Thanks for bearing with me on this 🙏

**Comment by aschonfeld** [comment]
These changes have now been released in v1.37.1
